tests/elementor_integration/Template/ControllerTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace TEC\Events\Integrations\Plugins\Elementor;
+
+use Tribe__Events__Main as TEC;
+use TEC\Events\Integrations\Plugins\Elementor\Template\Controller as Elementor_Template_Controller;
+
+class ControllerTest extends \Codeception\TestCase\WPTestCase {
+	public function setUp() {
+		parent::setUp();
+		tribe()->register( Elementor_Template_Controller::class );
+
+		// Create an event using The Events Calendar ORM before each test
+		$event_data = [
+			'title' => 'Sample Event',
+			'content' => 'This is a sample event description.',
+			'start_date' => '2024-09-23 01:00:00',
+			'end_date' => '2024-09-23 05:00:00',
+		];
+
+		$this->event_test = tribe_events()->set_args($event_data)->create();
+
+	}
+
+	public function tearDown(): void {
+		tribe( Elementor_Template_Controller::class )->unregister();
+		// Delete the event after each test
+		if ($this->event_test) {
+			tribe_events()->delete($this->event_test->ID);
+		}
+		parent::tearDown();
+	}
+
+	/**
+	 * It should allow the Elementor template override to be bypassed.
+	 *
+	 * @test
+	 */
+	public function should_allow_bypassing_override(): void {
+		// Set up filter to return true.
+		add_filter('tec_events_integration_elementor_bypass_template_override', function(){
+			return true;
+		});
```

Comment:
Ideally remove the filter after you are done, you could do this in your tearDown method or in any method marked in its doc as `@after`. So switch to `add_filter( 'tec_events_integration_elementor_bypass_template_override', '__return_true' )` so that in an `@after` method you can `remove_filter( 'tec_events_integration_elementor_bypass_template_override`, '__return_false`, 10 );`

Reply:
Wouldn't that be 
`remove_filter( 'tec_events_integration_elementor_bypass_template_override', '__return_true', 10 );`
As you are removing the `__return_true` function?

Reply:
Yes it would be `__return_true`.